### ISSUE TITLE
int added to str error changes based on order of desired output 

### ISSUE THREAD

**Issue by FearTheBadger** [mikefarah/yq#1234]
**Describe the bug**
I can get an int to add to the string of output if it is at the end of the output string. But if it is anywhere other than the end I get the `Error: !!int cannot be added to !!str`

Version of yq: 4.25.1
Operating system: linux
Installed via: binary release

**Input Yaml**
test.yml:
```yaml
ports:
  - name: test
    port: 80
```

**Command**

``` shell
yq '.. | select (. == "80") | parent | [.name + " " + .port]' test.yml
yq '.. | select (. == "80") | parent | [.port + " " + .name]' test.yml
```

**Actual behavior**

```shell
yq '.. | select (. == "80") | parent | [.name + " " + .port]' test.yml
- test 80
yq '.. | select (. == "80") | parent | [.port + " " + .name]' test.yml
Error: !!int cannot be added to !!str
```

**Expected behavior**

```shell
yq '.. | select (. == "80") | parent | [.name + " " + .port]' test.yml
- test 80
yq '.. | select (. == "80") | parent | [.port + " " + .name]' test.yml
- 80 test
```
*OR*
```shell
yq '.. | select (. == "80") | parent | [.name + " " + .port]' test.yml
Error: !!int cannot be added to !!str
yq '.. | select (. == "80") | parent | [.port + " " + .name]' test.yml
Error: !!int cannot be added to !!str
```

**Additional context**
None

**Comment by mikefarah** [comment]
Yeah not sure why I did that, will fix.

**Comment by mikefarah** [comment]
Fixed in v4.25.3
